File: DSAlgorithms/Leetcode/binary_search/assorted/c++/441-arranging_coins.cpp
class Solution {
public:
    int arrangeCoins(int n) {
       int left = 1;
       int right = n;
       int result = 0;
       while(left <= right) 
       {
           int mid = left + (right - left) / 2;
           long long needed = (double)mid * (mid + 1) /2;
           if(needed > n)
                right = mid - 1;
           else
           {
               left = mid + 1;
               result = std::max(result, mid);
           }
       }
       return result;
    }
};

File: dsa_algo/leetcode/ts/src/binary_search/441-arranging_coins.ts
const arrangeCoins = (n: number): number =>{
    let left = 1;
    let right = n
    let result = 0;
    while(left <= right)
    {
        let mid = Math.floor(left + (right - left) /2);
        let needed = (mid / 2) * (mid + 1);
        if(needed > n)
            right = mid - 1;
        else 
        {
            left = mid + 1;
            result = Math.max(result, mid)
        }
    }
    return result;
};